src/radical/pilot/task_manager.py

```diff
@@ -307,21 +311,18 @@ def _pilot_state_cb(self, pilots, state=None):
 
             if state in rps.FINAL:
 
-                self._log.debug('pilot %s is final', pilot.uid)
+                self._log.debug('pilot %s is final', pid)
 
-                # FIXME: MongoDB
-                # TODO: fail all non-final tasks which were assigned to that
-                # pilot
-                continue
+                tasks = list()
+                for task in self._tasks.values():
+
+                    task['exception']        = 'RuntimeError("pilot died")'
```

Comment:
IMHOP, it better to wrap `exception str` with an actual exception, like, for example `RuntimeError(pilot died)`

Reply:
Yes, that would be better.  Alas, exceptions are not generally serializable, and thus we only forward a string representation of it.    Changing that would involve some magic similar to what is done in the Parsl code base - that's out of scope at this point.